Data/Aeson/Types/ToJSON.hs

```diff
@@ -1633,6 +1631,43 @@ instance (ToJSON1 f, ToJSON1 g, ToJSON a) => ToJSON (Compose f g a) where
     toEncodingList = liftToEncodingList toEncoding toEncodingList
     {-# INLINE toEncodingList #-}
 
+
+instance (ToJSON1 f, ToJSON1 g) => ToJSON1 (Product f g) where
+    liftToJSON tv tvl (Pair x y) = liftToJSON2 tx txl ty tyl (x, y)
+      where
+        tx = liftToJSON tv tvl
+        txl = liftToJSONList tv tvl
+        ty = liftToJSON tv tvl
```

Comment:
the duplication is because they actually have different types (or then I'd need to give a type annotation, so the same four rows)